bugshaker/src/main/java/com/github/stkent/bugshaker/MainActivity.java

```diff
@@ -0,0 +1,217 @@
+			setButtonClickListener(getResources().getInteger(R.integer.medium_size), mediumBtn, brushDialog, false);
+			setButtonClickListener(getResources().getInteger(R.integer.large_size), largeBtn, brushDialog, false);
+
+			brushDialog.show();
+		}
+
+		else if (view.getId() == R.id.sendEmail) {
+			saveSendScreenshotAndLog();
+		}
+
+	}
+
+	public void paintClicked(View view) {
+		drawView.setBrushSize(drawView.getLastBrushSize());
+
+		if (view != currPaint) {
+			ImageButton imgView = (ImageButton) view;
+			String clickedButtonColor = view.getTag().toString();
+			drawView.setColor(clickedButtonColor);
+			imgView.setImageDrawable(getResources().getDrawable(R.drawable.paint_pressed));
+			currPaint.setImageDrawable(getResources().getDrawable(R.drawable.paint));
+			currPaint = (ImageButton) view;
+		}
+	}
+
+	private Uri getImageUri(Context inContext, Bitmap inImage) {
+		ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+		inImage.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
+		String path = MediaStore.Images.Media.insertImage(inContext.getContentResolver(), inImage, "Title", null);
+		return Uri.parse(path);
+	}
+
+	/*public Set<String> getValueStringSet(Context context) {
+		SharedPreferences settings;
+
+		settings = context.getSharedPreferences(PREFS_NAME2, Context.MODE_PRIVATE);
+		Set<String> text = (Set<String>) settings.getStringSet(PREFS_KEY2, null);
+		return text;
+	}*/
+
+	/*public String getValueString(Context context) {
+		SharedPreferences settings;
+		String text;
+		settings = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
+		text = settings.getString(PREFS_KEY, null);
+		return text;
+	}*/
+
+	private void saveSendScreenshotAndLog() {
+		AlertDialog.Builder sendDialog = new AlertDialog.Builder(this);
+
+		sendDialog.setTitle(getString(R.string.send_annotated_screenshot));
+		sendDialog.setMessage(getString(R.string.attach_annotated_screenshot_to_email));
+		sendDialog.setPositiveButton(getString(R.string.yes), new DialogInterface.OnClickListener() {
+			public void onClick(DialogInterface dialog, int which) {
+				drawView.setDrawingCacheEnabled(true);
+
+				String imgSaved = MediaStore.Images.Media.insertImage(
+					getContentResolver(), drawView.getDrawingCache(),
+					ScreenshotUtil.IMAGE_FILE_NAME, ScreenshotUtil.IMAGE_DESCRIPTION);
+				Bitmap screenshotBitmap = drawView.getDrawingCache();
+
+				Uri bitmapUri = getImageUri(getApplicationContext(), screenshotBitmap);
+
+				if (imgSaved != null) {
+					Toast savedToast = Toast.makeText(getApplicationContext(),
+						"Drawing saved to Gallery!", Toast.LENGTH_SHORT);
+
+					File log = LogcatUtil.saveLogcatToFile(getApplicationContext());
+					Uri logUri = Uri.fromFile(log);
+					Set<String> stringSet = SharedPreferencesUtil.getValueStringSet(getBaseContext());
+					String[] stringArray = stringSet.toArray(new String[stringSet.size()]);
```

Comment:
stringArray and stringSet don't tell the dev what we are fetching. Variable names should reflect what is actually contained in them, than being generic.